homeassistant/components/simplisafe/alarm_control_panel.py

```diff
@@ -137,10 +139,29 @@ def __init__(self, simplisafe: SimpliSafe, system: SystemType) -> None:
             system,
             additional_websocket_events=WEBSOCKET_EVENTS_TO_LISTEN_FOR,
         )
+        if code := self._simplisafe.entry.options.get(CONF_CODE):
+            if code.isdigit():
+                self._attr_code_format = CodeFormat.NUMBER
+            else:
+                self._attr_code_format = CodeFormat.TEXT
 
         self._last_event = None
         self._set_state_from_system_data()
 
+    @callback
+    def _is_code_valid(self, code: str | None, state: str) -> bool:
```

Comment:
This is not native device functionality?

Reply:
No, without this you can interact with simplisafe alarm without a pin, it does not get triggered, even if configured. 
More info in the issue: https://github.com/home-assistant/core/issues/124145

Reply:
Virtual functionality is not something we would allow.

Reply:
What is the purpose of configuring the pin in the integration configuration page?

![screen-20241019-104826_CuteGIF_20241019_104902.gif](https://github.com/user-attachments/assets/9bb23208-f5c6-4dde-b298-fb822775a9fa)

Reply:
> Virtual functionality is not something we would allow.

So is there nothing we can do here? 
This was operating like this for a long period, this PR simply enables that feature again. 
As you can see in the issue, there are a lot of users wanting this feature to be added back in. 
We could potentially update the user guide for this integration to explain how it works.